dwave/preprocessing/include/dwave/flags.hpp

```diff
@@ -19,9 +19,9 @@
 namespace dwave::presolve {
 
 enum Feasibility {
-    Unknown,     //< It is not known if the model is feasible or not
     Infeasible,  //< The model is known to be infeasible
     Feasible,    //< The model is known to be feasible
+    Unknown,     //< It is not known if the model is feasible or not
```

Comment:
Why the enum change? (just curious)

Reply:
Good spot. It's because by default the first value in enum gets assigned 0, so at some point if we want to do some operator overloading it would let us support
```c++
if (!presolver.feasibility()) {}
```
which will evaluate true only if the model is known to be infeasible. I haven't actually tested it and I think it currently won't work out-of-the-box, but makes it very slightly more future proof if we ever want to go there.

Reply:
I don't think that's a good idea. Enums shouldn't be interpreted as ints. (In your example `feasibility == unknown` would be interpreted as `feasible`)

Reply:
Well good thing that it doesn't work now then :shrug:. In either case, for presolve (where in general the model is assumed to be feasible until proven otherwise) treating unknown as feasible is actually desirable.

As I said, it makes it slightly more future proof **if** we want to ever go that direction.

Reply:
Ok, in that case, perhaps be explicit about the type, and the exact value each member has (instead of relying on implicit int zero for the first one; someone might accidentally reorder these members in the future, not realizing some external code is relying on the concrete int value associated).

Reply:
I am not sure what you mean, we don't rely on the implicit zero anywhere currently. I was also not sure it's a good idea, that's why I didn't follow through on adding the operator overloads we would need to actually make it happen.

So the reorder is just a hedge on the future, it may allow ourselves to add those operators without breaking binary compatibility by reordering the enum in the future. In that case we would of course document it.